core/src/main/java/org/mule/runtime/core/MuleMessageCorrelation.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.core;
+
+import static java.util.Optional.ofNullable;
+import static org.mule.runtime.core.routing.CorrelationMode.ALWAYS;
+import static org.mule.runtime.core.routing.CorrelationMode.IF_NOT_SET;
+import static org.mule.runtime.core.routing.CorrelationMode.NEVER;
+
+import org.mule.runtime.api.message.MuleMessage;
+import org.mule.runtime.core.routing.CorrelationMode;
+
+import java.io.Serializable;
+import java.util.Optional;
+
+/**
+ * Immutable container for correlation properties relative to a {@link MuleMessage}.
+ * 
+ * @since 4.0
+ */
+public class MuleMessageCorrelation implements Serializable
+{
+    private static final long serialVersionUID = -5687080761804624442L;
+
+    public static final String NOT_SET = "<not set>";
+
+    private final String id;
+    private final Integer groupSize;
+    private final Integer sequence;
+
+    /**
+     * Builds a new {@link MuleMessageCorrelation} with the given parameters.
+     * 
+     * @param id see {@link #getId()}.
+     * @param groupSize see {@link #getGroupSize()}.
+     * @param sequence see {@link #getSequence()}.
+     */
+    public MuleMessageCorrelation(String id, Integer groupSize, Integer sequence)
```

Comment:
Isn't it better to use two constructors, one that just takes id, and another that takes string and 2 int's (but not use Integer)?